js/index.js

```diff
@@ -1,10 +1,77 @@
 const isEqual = require("lodash.isequal");
 const expects = require("./expects.js");
 
+function getAllIndexes(arr, val) {
+  var indexes = [], i;
+  for (i = 0; i < arr.length; i++)
+    if (arr[i] === val)
+      indexes.push(i);
+  return indexes;
+}
+
+function checkVertical(puzzle, y, x, word) {
```

Comment:
For a function that returns a truthy value, can use an `isXYZ` naming format for clarity https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig

eg isVerticalMatch or something similar might improve clarity. For example it might be clearer when the function is used such as:

```
if (isVerticalMatch(puzzle, n, i, w)) {
  obj[w].push([n, i], [n + w.length - 1, i]);
}
```